### PR TITLE
DWR-705 Refactor migrate to remove the hardcoded schema names

### PR DESCRIPTION
**Pull request by agorina**
There are two changes:
1. removed schema names from SQLs
2. cleaned up test SQL files. I have noticed that some files had different names but pretty much the same content. More importantly my goal was to use the same test data in "step" tests as in the "batch" test.

### CODE REVIEW

File: group-processor/src/test/resources/WarehouseCodesSetup.sql [hunk 1]
```diff
@@ -2,18 +2,18 @@
...
-INSERT INTO warehouse_test.depth_of_knowledge  VALUES (-99,1,1,'Recall and Reproduction','something'),(-98,2,1,'Basic Skills and Concepts','anything');
-INSERT INTO warehouse_test.math_practice  VALUES (-99,'Make sense of problems and persevere in solving them', '-9'),(-98,'Reason abstractly and quantitatively', '-8');
-INSERT INTO warehouse_test.item_trait_score VALUES (-99,'Conventions-test'),(-98,'Evidence/Elaboration-test');
-INSERT INTO warehouse_test.target VALUES (-99, 'natural-99', -99,'10-1','WORD MEANINGS: Determine intended meanings of words..'),(-98, 'naturalId-99', -98,'10-12','WORD MEANINGS:..)');
-INSERT INTO warehouse_test.common_core_standard VALUES (-99, 'naturalId-99', 1,'ccommon core -99'),(-98, 'naturalId-98', 2, 'common core -98');
-INSERT INTO warehouse_test.item_difficulty_cuts  VALUES (-99, 1, 2, -99,-1.93882,-0.43906),(-98 ,1, 2, -98,-1.51022,0.14288);
-INSERT INTO warehouse_test.language VALUES (-99,'lan'),(-98,'gua');
-INSERT INTO warehouse_test.accommodation_translation VALUES (-99, -99,'Hola'),(-98, -98,'Hello');
\ No newline at end of file
+INSERT INTO grade (id, code, name) VALUES (-99,'99','K-garten'),(-98 ,'98','1st-grade');
```
Comment (agorina) [inline group-processor/src/test/resources/WarehouseCodesSetup.sql]: I am suspecting that it was an unintended "copy and paste" mistake here.

File: migrate-reporting/src/main/java/org/opentestsystem/rdw/ingest/migrate/reporting/configuration/DataSourceConfiguration.java [hunk 2]
```diff
@@ -26,7 +26,7 @@
      * @return DataSource
      */
     @Primary
-    @Bean(name = "datasource")
+    @Bean(name = "reportingDatasource")
```
Comment (agorina) [inline migrate-reporting/src/main/java/org/opentestsystem/rdw/ingest/migrate/reporting/configuration/DataSourceConfiguration.java]: It is really a reporting datasource, so I renamed it.

File: migrate-reporting/src/test/java/org/opentestsystem/rdw/ingest/migrate/reporting/repository/impl/JdbcReportingMigrateRepositoryIT.java [hunk 3]
```diff
@@ -14,14 +14,14 @@
...
-@Sql(statements = {"INSERT INTO reporting_test.migrate (id, job_id, status, first_at, last_at) VALUES" +
+@Sql(statements = {"INSERT INTO migrate (id, job_id, status, first_at, last_at) VALUES" +
         "(-10, 1, 20,'2017-07-18 19:45:33.966000', '2017-07-18 19:45:43.966000')," +
         "(-9, 2, 20, '2017-07-18 19:45:23.966000', '2017-07-18 19:45:33.966000');"
 })
 public class JdbcReportingMigrateRepositoryIT extends RepositoryIT {
 
     @Autowired
-    private JdbcTemplate jdbcTemplate;
+    private JdbcTemplate reportingJdbcTemplate;
```
Comment (agorina) [inline migrate-reporting/src/test/java/org/opentestsystem/rdw/ingest/migrate/reporting/repository/impl/JdbcReportingMigrateRepositoryIT.java]: "Reporting" is the default data source with its transaction manager, so no need to explicitly define it here or clean up the data after the tests.

File: migrate-reporting/src/test/java/org/opentestsystem/rdw/ingest/migrate/reporting/repository/impl/JdbcWarehouseImportRepositoryIT.java [hunk 4]
```diff
@@ -6,12 +6,19 @@
...
+import org.springframework.test.context.jdbc.SqlConfig;
+import org.springframework.transaction.annotation.Transactional;
 
 import java.time.Instant;
 
 import static org.assertj.core.api.Assertions.assertThat;
 
+@ActiveProfiles("warehouseTxtMgr")
+@Import({TestWarehouseDataSourceConfiguration.class})
+@Transactional(transactionManager = "warehouseTxManager")
```
Comment (agorina) [inline migrate-reporting/src/test/java/org/opentestsystem/rdw/ingest/migrate/reporting/repository/impl/JdbcWarehouseImportRepositoryIT.java]: I did not want to clean up the data after these tests.
Some of them do "DELETE" all, so I wanted it to be within a transaction that is rolled backet at the end of the tests. That is why I defined a warehouse transaction manager just for the test.

File: migrate-reporting/src/test/java/org/opentestsystem/rdw/ingest/migrate/reporting/SpringBatchStepIT.java [hunk 5]
```diff
@@ -12,7 +12,7 @@
 
     protected final Instant firstAt = Instant.parse("2017-07-18T19:05:34.965Z");
     protected final Instant lastAt = Instant.parse("2017-07-18T20:14:34.000Z");
-    protected static final String StagingTablePrefix = "reporting_test.staging_";
+    protected static final String StagingTablePrefix = "staging_";
```
Comment (malaffoon) [inline migrate-reporting/src/test/java/org/opentestsystem/rdw/ingest/migrate/reporting/SpringBatchStepIT.java]: I'm not convinced this adds value. Actually i think it makes things less readable. The table names are the table names; the fact that they have a consistent prefix is irrelevant. Personally, i would get rid of this and just have the table names. Unless i'm missing something?

File: migrate-reporting/src/test/java/org/opentestsystem/rdw/ingest/migrate/reporting/step/StageExamIT.java [hunk 6]
```diff
@@ -25,37 +26,35 @@
...
-        tableTestCounts.add(new TableTestCountHelper(jdbcTemplate, StagingTablePrefix + "exam_student", 0, "id in (-18)"));
-        tableTestCounts.add(new TableTestCountHelper(jdbcTemplate, StagingTablePrefix + "exam_available_accommodation", 3, "exam_id in (-87, -86, -85)"));
-        tableTestCounts.add(new TableTestCountHelper(jdbcTemplate, StagingTablePrefix + "exam_available_accommodation", 0, "exam_id in (-88)"));
-        tableTestCounts.add(new TableTestCountHelper(jdbcTemplate, StagingTablePrefix + "exam_item", 3, "exam_id in (-87, -86,  -85)"));
-        tableTestCounts.add(new TableTestCountHelper(jdbcTemplate, StagingTablePrefix + "exam_item", 0, "exam_id in (-88)"));
-        tableTestCounts.add(new TableTestCountHelper(jdbcTemplate, StagingTablePrefix + "exam_claim_score", 3, "exam_id in (-87, -86, -85)"));
-        tableTestCounts.add(new TableTestCountHelper(jdbcTemplate, StagingTablePrefix + "exam_claim_score", 0, "exam_id in (-88)"));
+        tableTestCounts.add(new TableTestCountHelper(reportingJdbcTemplate, StagingTablePrefix + "exam", 4, "id in (-88, -87, -86, -85)"));
```
Comment (malaffoon) [inline migrate-reporting/src/test/java/org/opentestsystem/rdw/ingest/migrate/reporting/step/StageExamIT.java]: To follow-up these would just be table names without concatenation, e.g. "staging_exam_student".

File: migrate-reporting/src/test/java/org/opentestsystem/rdw/ingest/migrate/reporting/SpringBatchStepIT.java (hunk 5 above)
Comment (agorina) [reply migrate-reporting/src/test/java/org/opentestsystem/rdw/ingest/migrate/reporting/SpringBatchStepIT.java]: I agree. But this review is big as is. Will do it as a separate PR.

**Pull request merged**